crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -431,15 +421,114 @@ input CheckpointId {
 """
 Some 0x2::coin::Coin Move object.
 """
-type Coin {
+type Coin implements IMoveObject & IObject & IOwner {
+	address: SuiAddress!
 	"""
-	Balance of the coin object
+	Objects owned by this object, optionally `filter`-ed.
 	"""
-	balance: BigInt
+	objects(first: Int, after: String, last: Int, before: String, filter: ObjectFilter): MoveObjectConnection!
+	"""
+	Total balance of all coins with marker type owned by this object. If type is not supplied,
+	it defaults to `0x2::sui::SUI`.
+	"""
+	balance(type: String): Balance
+	"""
+	The balances of all coin types owned by this object.
+	"""
+	balances(first: Int, after: String, last: Int, before: String): BalanceConnection!
+	"""
```

Comment:
its kinda crazy that a coin can own other coins lol

Reply:
i feel like the incredible flexibility of sui's object model was not clear to me until seeing this. and of course in no small part due to the transfer-to-object work

Reply:
Actually a coin could own other coins even before transfer to object -- it just couldn't do anything with those coins 😅. Also, yeah, this is probably not something that anyone should be doing in practice.